packages/itwinui-react/src/utils/components/FocusTrap.tsx

```diff
@@ -19,37 +18,55 @@ export type FocusTrapProps = {
  */
 export const FocusTrap = (props: FocusTrapProps) => {
   const { children } = props;
-  const childRef = React.useRef<HTMLElement>(undefined);
 
-  const getFirstLastFocusables = () => {
-    const elements = getTabbableElements(childRef.current);
+  // Get childRef via firstFocusTrapRef instead of by passing ref to cloneElementWithRef(children) to avoid re-renders.
+  // https://github.com/iTwin/iTwinUI/issues/2445
+  const firstFocusTrapRef = React.useRef<HTMLDivElement>(null);
+  const childrenElement = firstFocusTrapRef.current?.nextElementSibling as
```

Comment:
Refs (and DOM APIs) should not be read during render, so you'll want to move this in the callback or into an Effect.